gapil/resolver/flow.go

```diff
@@ -20,13 +20,18 @@ type visitedFuncs map[*semantic.Function]bool
 
 // resolveFenceOrder analyses comamnds and subroutines for the placement of the
 // fence.
-func resolveFenceOrder(rv *resolver, f *semantic.Function, visitedFuncs visitedFuncs) semantic.LogicalOrder {
+func resolveFenceOrder(rv *resolver, f *semantic.Function, visitedFuncs visitedFuncs, preFence bool, recursive *bool) semantic.LogicalOrder {
 	if f.Order.Resolved() {
 		return f.Order // Already resolved
 	}
 	if visitedFuncs[f] {
-		rv.errorf(f, "Cyclic dependeny found")
-		return 0
+		if !preFence {
```

Comment:
I'm really struggling to get my head wrapped around this. Why do you need this error here, won't we catch the fence in recursive functions with the `"Fence in recursive function"` error below?

Reply:
That is true, we can remove that clause, and just handle it elsewhere.